src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -12,84 +13,126 @@ using namespace Microsoft::Terminal::Core;
 // - A vector of rectangles representing the regions to select, line by line. They are absolute coordinates relative to the buffer origin.
 std::vector<SMALL_RECT> Terminal::_GetSelectionRects() const
 {
-    std::vector<SMALL_RECT> selectionArea;
+    std::vector<SMALL_RECT> result = {};
 
     if (!IsSelectionActive())
     {
-        return selectionArea;
+        return result;
     }
 
+    try
+    {
+        // NOTE: (0,0) is the top-left of the screen
+        // the physically "higher" coordinate is closer to the top-left
+        // the physically "lower" coordinate is closer to the bottom-right
+        auto [higherCoord, lowerCoord] = _PreprocessSelectionCoords();
+
+        SHORT selectionRectSize;
+        THROW_IF_FAILED(ShortSub(lowerCoord.Y, higherCoord.Y, &selectionRectSize));
+        THROW_IF_FAILED(ShortAdd(selectionRectSize, 1, &selectionRectSize));
+
+        std::vector<SMALL_RECT> selectionArea = {};
+        selectionArea.reserve(selectionRectSize);
+        for (auto row = higherCoord.Y; row <= lowerCoord.Y; row++)

```

Comment:
```suggestion
        for (const auto row = higherCoord.Y; row <= lowerCoord.Y; row++)
```

Reply:
Can't do this one. Incrementing row at the end of the line.